engine/src/execution_graph/logic_controllers/BatchAggregationProcessing.h

```diff
@@ -316,7 +316,7 @@ class MergeAggregateKernel : public kernel {
                         concatenated->toBlazingTableView(), mod_group_column_indices);
             } else if (group_column_indices.size() == 0) {
                 // aggregations without groupby are only merged on the master node
-                if(context->isMasterNode(ral::communication::CommunicationData::getInstance().getSelfNode())) {
+                if( (context->getTotalNodes() == 1) ||  (context->isMasterNode(ral::communication::CommunicationData::getInstance().getSelfNode()))  ) {
```

Comment:
This looks like a band-aid. We need to instead figure out why if there is only one node, its not automatically the MasterNode